src/agents/agent.py

```diff
@@ -63,7 +64,16 @@ def search_queries(self, queries: list, project_name: str) -> dict:
         results = {}
         
         knowledge_base = KnowledgeBase()
-        bing_search = BingSearch()
+        user_search_engine = Config().get_search_engine()
+
+        if user_search_engine == "DDGS":
+            search_engine = DuckDuckGoSearch()
+        else:
+            """
+            Set default search engine to Bing
+            """
+            search_engine = BingSearch()
```

Comment:
**Optimization Issue**: The implementation for choosing the search engine based on user preference (Bing or DuckDuckGo) every time a search query is made can lead to inefficiency, especially in scenarios where multiple queries are being processed in a loop. This approach results in repeated checks of the user's preferred search engine, which is unnecessary after the first determination has been made. <br> **Fix**: Initialize the search engine once at the start of the application or when the user's preference is updated. Store this instance for reuse in all subsequent searches to avoid redundant checks and object instantiations. This can be achieved by implementing a factory pattern that initializes the correct search engine instance based on the user's preference and provides this instance wherever needed. <br> **Code Suggestion**: 
 ```
 +class SearchEngineFactory:
 +    @staticmethod
 +    def get_search_engine(engine_preference):
 +        if engine_preference == 'DDGS':
 +            return DuckDuckGoSearch()
 +        else:
 +            return BingSearch()
 
  def search_queries(self, queries: list, project_name: str) -> dict:
      results = {}
      knowledge_base = KnowledgeBase()
 -    bing_search = BingSearch()
 +    user_search_engine = Config().get_search_engine()
 +    search_engine = SearchEngineFactory.get_search_engine(user_search_engine)
      browser = Browser()

 ```